dds/idl/be_produce.cpp

```diff
@@ -271,6 +271,22 @@ void postprocess(const char* fn, ostringstream& content,
 
   out << be_global->get_include_block(which);
 
+  if (which == BE_GlobalData::STREAM_CPP) {
+    // For Unity builds, names in anonymous namespace can clash, so this is used
+    // to make namespaces with names based on the file instead.
+    std::string prefix = to_macro(fn);
+    for (size_t i = 0; i < prefix.size(); ++i) {
+      prefix[i] = tolower(prefix[i]);
+    }
+    out <<
+      "\n"
+      "#ifdef OPENDDS_IDL_FILE_SPECIFIC\n"
+      "#  undef OPENDDS_IDL_FILE_SPECIFIC\n"
+      "#endif\n"
+      "#define OPENDDS_IDL_FILE_SPECIFIC(base, index) " << prefix << "##_##base##index\n"
+      "\n";
+  }
```

Comment:
Instead of generating code for the preprocessor to determine the name, how about generating that name directly in the be_*?

Reply:
I think tried that first, but went with this because it seemed slightly cleaner for the code in opendds_idl.

Reply:
Maybe at some point (not this PR) we'll move it to a common header so it can be used in other places.

Reply:
Another option is [`UNITY_BUILD_UNIQUE_ID`](https://cmake.org/cmake/help/latest/prop_tgt/UNITY_BUILD_UNIQUE_ID.html#prop_tgt:UNITY_BUILD_UNIQUE_ID).

Reply:
#4307